homeassistant/components/yolink/__init__.py

```diff
@@ -80,6 +92,28 @@ def on_message_callback(message: tuple[str, BRDP]) -> None:
         device_coordinator = device_coordinators.get(device_id)
         if device_coordinator is None:
             return
+        if device_coordinator.device.device_type == ATTR_DEVICE_SMART_REMOTE:
+            device_registry = dr.async_get(hass)
+            device_entry = device_registry.async_get_device(
+                identifiers={(DOMAIN, device_coordinator.device.device_id)}
+            )
+            if device_entry is not None:
+                btn_press_event = resolved_state.get("event")
+                if btn_press_event is not None:
```

Comment:
We can combine these two checks if we use a walrus operator for `btn_press_event`.